conan/internal/runner/ssh.py

```
@@ -119,92 +109,97 @@ def ensure_runner_environment(self):
+        result = self.remote_conn.run_command(f"{python_command} -m pip install conan{f'=={version}' if version != 'dev' else ' --upgrade'}", "Installing conan")
+        if not result.success:
+            self.output.error(f"Unable to install conan in venv: {result.stderr}")
 
+    def _create_remote_conan_wrapper(self, remote_conan_home: str, remote_folder: str, conan_cmd: str):
+
+        # Create conan wrapper with proper environment variables
         remote_env = {
             'CONAN_HOME': remote_conan_home,
-            'CONAN_RUNNER_ENVIRONMENT': "1"
+            'CONAN_RUNNER_ENVIRONMENT': "1"  # This env will prevent conan (running remotely) to start an infinite remote call
         }
-        if is_windows:
+        if self.is_remote_windows:
             # Wrapper script with environment variables preset
-            env_lines = "\n".join([f"set {k}={v}" for k,v in remote_env.items()])
-            conan_bat_contents = f"""@echo off\n{env_lines}\n{conan_cmd} %*\n"""
-            conan_bat = remote_folder + "/conan.bat"
-            try:
-                sftp = self.client.open_sftp()
-                sftp.putfo(BytesIO(conan_bat_contents.encode()), conan_bat)
-            except:
-                raise ConanException("unable to set up Conan remote script")
-            finally:
-                sftp.close()
-
-            self.remote_conan = conan_bat
-        _, _stdout, _stderr = self.client.exec_command(f"{self.remote_conan} config home")
-        ssh_info(f"Remote conan config home returned: {_stdout.read().decode().strip()}")
-        _, _stdout, _stderr = self.client.exec_command(f"{self.remote_conan} profile detect --force")
-        self._copy_profiles()
+            env_lines = "\n".join([f"set {k}={v}" for k, v in remote_env.items()])
+            conan_wrapper_contents = f"""@echo off\n{env_lines}\n{conan_cmd} %*\n"""
+        else:
+            env_lines = "\n".join([f"export {k}={v}" for k, v in remote_env.items()])
```

We need to document in the comments why this is needed in the first place and why we don't pass the environment directly via paramiko - IIRC on Windows the server silently ignores environment variables

but also need to document why it's needed on non-windows hosts